iree/base/internal/BUILD

```diff
@@ -341,6 +324,38 @@ cc_test(
     ],
 )
 
+#===------------------------------------------------------------------------===#
+# Utilities with thread dependencies
+#===------------------------------------------------------------------------===#
+
+iree_cmake_extra_content(
+    content = """
+if (NOT ${IREE_ENABLE_THREADING})
```

Comment:
let's leave a note about why status needs threading to make future cleanup easier
also nit: style

```suggestion
# TODO(#3848): absl has a thread dependency, move this condition after status.
if(NOT ${IREE_ENABLE_THREADING})
```

Reply:
Done.